crates/sui-json-rpc/src/governance_api.rs

```diff
@@ -38,6 +38,72 @@ impl GovernanceReadApi {
             .await?)
     }
 
+    async fn get_stakes(&self, staked_sui_id: Vec<ObjectID>) -> Result<Vec<DelegatedStake>, Error> {
+        let stakes = futures::future::try_join_all(
+            staked_sui_id
+                .iter()
+                .map(|id| self.state.get_move_object::<StakedSui>(id)),
+        )
+        .await?;
+        if stakes.is_empty() {
+            return Ok(vec![]);
+        }
+
+        let pools = stakes
```

Comment:
Most of this logic are same `get_delegated_stake`, can we refactor this to share the same code? Will make it easier to maintain.